samples/cpp/hello_nv12_input_classification/main.cpp

```diff
@@ -54,258 +62,94 @@ std::pair<size_t, size_t> parseImageSize(const std::string& size_string) {
                       << std::endl;
             return EXIT_FAILURE;
         }
 
-        const std::string input_model{argv[1]};
-        const std::string input_image_path{argv[2]};
-        size_t input_width = 0, input_height = 0;
+        const std::string model_path{argv[1]};
+        const std::string image_path{argv[2]};
+        size_t input_width = 0;
+        size_t input_height = 0;
         std::tie(input_width, input_height) = parseImageSize(argv[3]);
         const std::string device_name{argv[4]};
         // -----------------------------------------------------------------------------------------------------
 
-        // ------------------------------ Read image names
-        // -----------------------------------------------------
-        auto image_names = readInputFileNames(input_image_path);
-
-        if (image_names.empty()) {
-            throw std::invalid_argument("images not found");
-        }
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- Step 1. Initialize inference engine core
-        // ------------------------------------------------
-        Core ie;
-        // -----------------------------------------------------------------------------------------------------
-
-        // Step 2. Read a model in OpenVINO Intermediate Representation (.xml and
-        // .bin files) or ONNX (.onnx file) format
-        CNNNetwork network = ie.ReadNetwork(input_model);
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- Reshape model
-        // -------------------------------------------------
-        size_t netInputSize = isBatchedBlobSupported(ie, device_name) ? image_names.size() : 1;
-        ICNNNetwork::InputShapes inputShapes = network.getInputShapes();
-        for (auto& shape : inputShapes) {
-            auto& dims = shape.second;
-            if (dims.empty()) {
-                throw std::runtime_error("Network's input shapes have empty dimensions");
-            }
-            dims[0] = netInputSize;
-        }
-        network.reshape(inputShapes);
-        size_t batchSize = network.getBatchSize();
-        std::cout << "Batch size is " << batchSize << std::endl;
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- Step 3. Configure input and output
-        // -------------------------------------------
-        // --------------------------- Prepare input blobs
-        // -----------------------------------------------------
-        if (network.getInputsInfo().empty()) {
-            std::cerr << "Network inputs info is empty" << std::endl;
-            return EXIT_FAILURE;
+        // -------- Read image names --------
```

Comment:
Previously nv12 sample, in addition to single file, could accept folder with NV12 images, now it accepts single file only. Is it expected?

Reply:
@nosovmik yes, remember, we would like to focus samples on using API rather then provide them tool to validate different scenarios of the same API usage. We cover batching in another sample